feat(exporter/syslog)!: replace syslog exporter with syslog exporter from opentelemetry-collector-contrib (#1341)
SumoLogic/sumologic-otel-collector: docs/upgrading.md

    - [Removing unnecessary metadata using the resourceprocessor](#removing-unnecessary-metadata-using-the-resourceprocessor)
    - [Moving record-level attributes used for metadata to the resource level](#moving-record-level-attributes-used-for-metadata-to-the-resource-level)

## Upgrading to v0.90.0-sumo-0

### Change configuration for `syslogexporter`

> To migrate, change keys in configuration for `syslogexporter`, replace:

Reviewer: ```suggestion
To migrate, rename the following keys in configuration for `syslogexporter`:
```

> - `protocol` key with `network` key

Reviewer: ```suggestion
- rename `protocol` property to `network`
```

> - `format` key with `protocol`

Reviewer: ```suggestion
- rename `format` property to `protocol`
```

For example, given the following configuration:

```yaml

> syslog:

Reviewer 2: ```suggestion
  syslog:
```

Author: fixed
